### PR TITLE
Event 12

### PR DESCRIPTION
**Pull request by mazvydasb**

### CODE REVIEW

File: Sources/EventsTracker/Event+SDKEvents.swift [hunk 1]
```diff
@@ -296,6 +343,16 @@ extension Event {
...
     }
     
+    public static func basicAnalytics(
+        category: String,
+        action: String,
+        appVersion: String,
+        screenName: String?,
+        label: String?
+        ) -> Event {
+        return basicAnalytics(category: category, action: action, appVersion: appVersion, screenName: screenName, label: label)
```
Comment (lozhuf) [inline Sources/EventsTracker/Event+SDKEvents.swift]: Because the function signature (with default values applied) of the internal version of this func (line 72) is the same as the public version, it is causing a warning (which may be a runtime error): `Event+SDKEvents.swift:346:24: warning: all paths through this function will call itself`

The fix for this is to specify some of the defaulted properties (the timestamp is easiest), so that it knows which function you are meaning to call. eg.
```
return basicAnalytics(category: category, action: action, appVersion: appVersion, screenName: screenName, label: label, timestamp: Date())
```

File: Sources/EventsTracker/Event+SDKEvents.swift [hunk 2]
```diff
@@ -58,6 +59,52 @@ extension Event {
             .addingViewToken(content: publicationId.rawValue, tokenizer: tokenizer)
     }
     
+    /**
+     The event to report basic analytics about new sessions and views opened.
```
Comment (lozhuf) [inline Sources/EventsTracker/Event+SDKEvents.swift]: Not just limited to these cases, so maybe reword as `... to report basic analytics **such** as new sessions ...`

File: Sources/EventsTracker/Event+SDKEvents.swift [hunk 3]
```diff
@@ -58,6 +59,52 @@ extension Event {
...
+     - parameter screenName: Name of the view currently being presented.
```
Comment (lozhuf) [inline Sources/EventsTracker/Event+SDKEvents.swift]: Param ordering needs to be updated

File: Sources/EventsTracker/Event+SDKEvents.swift [hunk 4]
```diff
@@ -58,6 +59,52 @@ extension Event {
...
+            payload["s"] = .string(name)
+        }
+        
+        if let lbl = label {
+            payload["l"] = .string(lbl)
+        }
+        
+        let viewTokenContent: String = [category, action, label, screenName]
+            .compactMap{ $0 }
+            .joined()
```
Comment (lozhuf) [inline Sources/EventsTracker/Event+SDKEvents.swift]: I think it might be worthwhile using a separator, like `.` or something. This just catches the (probably unlikely, but still possible) case where different category and action strings result in the same final string. Totally made up example:
`view` + `input` = `viewinput` or `view.input`
`viewin` + `put` = `viewinput` or `viewin.put`

**Comment by mortenbo** [comment]
@mazvydasb how do you make the view token? We need to use the client ID as well. And where do you get that from?

**Comment by lozhuf** [comment]
This is all handled automatically by the eventsTracker sdk.
